lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -93,6 +88,26 @@ def load
         cookbook_settings
       end
 
+      # Load the cookbook. Does not raise an error if given a non-cookbook
+      # directory as the cookbook_path. This behavior is provided for
+      # compatibility, it is recommended to use #load! instead.
+      def load
+        Chef::Log.warn "This method is deprecated. Use load! instead"
+        # metadata # force lazy evaluation to occur
+
+        # re-raise any exception that occurred when reading the metadata
+      #   raise_metadata_error!
+
+      #   load_all_files
+
+      #   remove_ignored_files
+
+      #   if empty?
+      #     Chef::Log.warn "Found a directory #{cookbook_name} in the cookbook path, but it contains no cookbook files. skipping."
+      #   end
+      #   cookbook_settings
```

Comment:
Ah what you should probably do here is just force it to `raise` instead of warn and hard error out.

I think I can see what its going to do though.  It looks like this is mostly used by the cookbook_version_loader and by `chef_fs/file_system/repository/chef_repository_file_system_cookbook_dir.rb` (I think).  If you switch that to a raise, and then convert those calls from `load` to `load!` I think it'll be close to passing.